R/expect-no-condition.R

```diff
@@ -0,0 +1,113 @@
+#' Does code running without error, warning, message, or other condition?
+#'
+#' These expectations assert the opposite of [expect_error()],
+#' `expect_warning()`, `expect_message()`, and `expect_condition()`, namely
+#' that the code runs without error, warning, or message respectively.
+#'
+#' @inheritParams expect_error
+#' @param message,class The default usage with `message = NULL, clas = NULL`
```

Comment:
This is a hard one. On the one hand, we want users to test for specific warnings. On the other hand, we don't want them to test for messages that are most probably generated in other packages.

This makes me think that `expect_warning(x, NA)` should be replaced by snapshots?

Reply:
I've tweaked the text to reinforce that you should be using `message` for your own conditions.